lib/chef/resource/macports_package.rb

```diff
@@ -16,10 +16,11 @@
 # limitations under the License.
 #
 
+require 'chef/resource/package'
+
 class Chef
   class Resource
     class MacportsPackage < Chef::Resource::Package
-      provides :package, os: "darwin"
```

Comment:
Can you elaborate?  I'm super unclear what is supposed to act differently when the `provides` statements are in there, given that you have to require `chef/resource/package` to compile this, and that file includes the set_resource_priority_map.

The confusion might be in what `def self.provides?` does?  If you call `Chef::Resource::MacportsPackage.provides?(<os x node object>, :package)`, it will go to the `ResourcePriorityMap` to find out whether it is provided or not (it lists all resource classes matching the node for `package`, and checks to see if MacportsPackage is one of them). So the priority array from package will actually affect this, and `provides?` will give a correct result.